app/controllers/results_controller.rb

```diff
@@ -653,8 +653,8 @@ def add_extra_mark
   end
 
   def remove_extra_mark
-    extra_mark = ExtraMark.find(params[:id])
-    result = extra_mark.result
+    result = Result.find(params[:id])
+    extra_mark = ExtraMark.find(params[:extra_mark_id])
```

Comment:
We can be a bit better with using associations, something like `result.extra_marks.find(...)`, so that only the extra marks associated with that result are searched.

(This prevents a user putting in a URL with an unrelated result id and extra mark id.)